app/controllers/sign_up/recovery_codes_controller.rb

```diff
@@ -6,7 +6,11 @@ class RecoveryCodesController < ApplicationController
     before_action :confirm_has_not_already_viewed_recovery_code
 
     def show
-      user_session.delete(:first_time_recovery_code_view)
+      if user_session[:first_time_recovery_code_view].present?
+        @show_progress_bar = true
+        user_session.delete(:first_time_recovery_code_view)
```

Comment:
I was thinking there had to be a prettier way to do that, I will try it out, thank you for the suggestions!